instrumentation/storage-test/src/main/java/io/opentelemetry/instrumentation/storage/StorageInstrumenter.java

```diff
@@ -0,0 +1,87 @@
+package io.opentelemetry.instrumentation.storage;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.auto.bootstrap.InstrumentationContext;
+import io.opentelemetry.auto.tooling.Instrumenter;
+import io.opentelemetry.auto.tooling.Instrumenter.Default;
+import io.opentelemetry.context.Scope;
+import java.util.HashMap;
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.asm.Advice.Argument;
+import net.bytebuddy.asm.Advice.Origin;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import static io.opentelemetry.context.ContextUtils.withScopedContext;
+import static io.opentelemetry.instrumentation.storage.StorageDecorator.DECORATE;
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+@AutoService(Instrumenter.class)
+public class StorageInstrumenter extends Default {
+
+  public StorageInstrumenter() {
+    super("storage");
+  }
+
+  @Override
+  public ElementMatcher<? super TypeDescription> typeMatcher() {
+    return named("io.opentelemetry.benchmark.storage.StorageBenchmark");
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    Map result = new HashMap();
+    result.put(named("contextInternal"), getClass().getName() + "$GrpcContextAdvice");
+    result.put(named("fieldInternal"), getClass().getName() + "$FieldAdvice");
+    return result;
+  }
+
+  @Override
+  public String[] helperClassNames() {
+    return new String[] {
+        packageName + ".StorageDecorator",
+    };
+  }
+
+  @Override
+  public Map<String, String> contextStore() {
+    return singletonMap("java.lang.String", Integer.class.getName());
```

Comment:
this is never going to use the field-based approach, because we can at most retransform `java.lang.String`, at which time we cannot modify its structure.

also, even if we could add a field to `java.lang.String`, that would add extra memory requirement to so many String instances. and assuming that we didn't need to map all of them, it likely would perform worse than maintaining a weak map of only the `String` instances that we need to map

Reply:
First, are you saying that performance of `InstrumentationContext` depends on the target class?

Second, how can I make sure from the logs or any other means, that in my benchmark it does not use field-based approach?

Reply:
> First, are you saying that performance of `InstrumentationContext` depends on the target class?

`java.lang.String` was just an unfortunate example to pick, as it typically has the most instances of any class in memory, so adding a field to it would have large repercussions (although we can't add a field to it, so it's a doubly unfortunate example to pick?)

> Second, how can I make sure from the logs or any other means, that in my benchmark it does not use field-based approach?

This is a good question, and I have two answers, neither of which you're going to like
* take a heap dump and comb through the reference graph
* run with `-Dnet.bytebuddy.dump=/some/folder` and comb through the bytecode

Ask me how I know this 😭 (https://github.com/microsoft/ApplicationInsights-Java/issues/1258)

Reply:
I should add, there's probably a better way, but I didn't find it

Reply:
Completely pointless trivia but I guess on Java 8 `String` has enough space for another field, but not on 9+, if I remember correctly they could store the isAscii aspect for free.